src/datasources/DataSource_Application.ts

```diff
@@ -205,8 +204,8 @@ export class DataSourceApplication {
       const data: any = {
         group: group,
         timeFrame: {
-          to: timeFilter.to,
-          windowSize: windowSize,
+          to: floorToGranularity(timeFilter.to, metric.granularity),
+          windowSize: ceilToGranularity(windowSize, metric.granularity),
```

Comment:
Shouldn't the same adjustment be applied also on line 258 for `/api/application-monitoring/metrics/applications`?

Reply:
yes, good catch